pytorch_forecasting/metrics/tests/test_all_metrics.py

```diff
@@ -0,0 +1,458 @@
+    "weighted_2d",
+    "weighted_3d",
+    "distribution",
+    "classification",
+]
+
+
+@pytest.fixture(params=get_all_metrics())
+def metric_class(request):
+    """Fixture to provide metric classes."""
+    return request.param
+
+
+@pytest.fixture
+def metric_instance(metric_class):
+    if issubclass(metric_class, (MultiLoss, CompositeMetric, TorchMetricWrapper)):
+        pytest.skip(f"{metric_class.__name__} requires special instantiation.")
+
+    try:
+        return metric_class()
+    except Exception as e:
+        pytest.fail(f"Failed to instantiate {metric_class.__name__}: {e}")
+
+
+@pytest.fixture(scope="session")
+def test_data():
+    """Session-scoped fixture for test data."""
+    return get_test_data()
+
+
+class TestAllPtMetrics:
+    """Test suite for all metrics in PyTorch Forecasting.
+
+    This test class implements a comprehensive testing strategy for metrics that:
+
+    1. Tests all metrics against diverse input formats (point forecasts, quantiles,
+    packed sequences, weighted inputs, distributions, classification)
+
+    2. Uses pytest.skip to gracefully handle inherent incompatibilities between
+    metrics and data formats, recognizing that not all metrics work with all data
+    types. This approach distinguishes between expected incompatibilities vs actual
+    bugs.
+
+    3. Validates both the functional interface (direct calls) and stateful interface
+    (update/compute pattern) of metrics.
+
+    4. Checks important metric properties including:
+
+    - Output tensor validity (finite values, correct shape)
+    - Reduction modes behavior (none, mean, sqrt-mean)
+    - Packed sequence handling for variable-length data
+    - Special methods like to_prediction and to_quantiles
+    - Metric composition behavior
+    - Example validation through doctests
+
+    The test design favors comprehensiveness over specificity, attempting all
+    combinations rather than hardcoding expected compatibilities. This makes the
+    tests more maintainable and future-proof, automatically handling new metrics
+    or data formats without requiring test updates.
+
+    When encountering incompatibilities, tests are skipped with descriptive messages
+    rather than marked as failures, since incompatibility is treated as
+    expected behavior and not a bug. This approach results in many skipped tests
+    but ensures genuine issues are visible and all metrics receive appropriate
+    validation against compatible formats.
+
+    Note
+    ----
+    There is a plan to implement a variable counter in each test to track if the
+    metric accepts atleast a single data format. This will help in
+    identifying metrics that are not tested at all, and thus not compatible with
+    any provided data format, and raise an error.
+    """
+
+    def test_metric_instantiation(self, metric_class):
+        "Test instantiation of metric classes."
+        try:
+            if issubclass(metric_class, (MultiLoss, CompositeMetric)):
```

Comment:
I get that we need this right now - from a design perspective, stepouts like this should be avoided in the tests. Maybe not now, but we should think about how to remove the stepouts later. Tags tend to be a good option.